matplotcheck/raster.py

```diff
@@ -114,19 +112,65 @@ def assert_legend_accuracy_classified_image(
         Finally those two arrays of strings are compared. Passes if they match.
         """
         # Retrieve image array
-        im_data = []
-        if self.ax.get_images():
-            im = self.ax.get_images()[0]
-            im_data, im_cmap = im.get_array(), im.get_cmap()
+        im_data = self.get_plot_image()
+        im = self.ax.get_images()[0]
+        im_cmap = im.get_cmap()
+
         assert list(im_data), "No Image Displayed"
 
+        # Retrieve legend entries and find which element of all_label_options
+        # matches that entry
+        legend_dict = self.get_legend_labels(all_label_options)
```

Comment:
why does this need all_label_options (and then let's rename all_label_options to `expected_labels`